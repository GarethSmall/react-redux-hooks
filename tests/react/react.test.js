// @flow
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { MockApp } from '../__fixtures__/react/MockApp';
const { describe, test, expect } = global;


/**
 * Tests:
 */
describe('react', () => {
  test('should render', () => {
    let component = ReactTestRenderer.create(<MockApp />);
    const button = component.root.find(el => el.props.id === 'fetch-users');
    button.props.onClick();
    component.update(<MockApp />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
