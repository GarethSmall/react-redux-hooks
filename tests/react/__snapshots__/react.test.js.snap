// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react should render 1`] = `
Array [
  <input
    id="username"
    onChange={[Function]}
    type="text"
    value=""
  />,
  <button
    id="create-users"
    onClick={[Function]}
  >
    Create User
  </button>,
  <button
    id="fetch-users"
    onClick={[Function]}
  >
    Fetch Users
  </button>,
  <ul>
    <li>
      Full Name: 
      Joey Tribbiani
    </li>
    <li>
      Full Name: 
      Monica Geller
    </li>
    <li>
      Full Name: 
      Chandler Bing
    </li>
    <li>
      Full Name: 
      Phoebe Buffay
    </li>
    <li>
      Full Name: 
      Rachel Green
    </li>
    <li>
      Full Name: 
      Ross Geller
    </li>
  </ul>,
  <div>
    Yahoo!
  </div>,
]
`;
